articles/nodejs-structural-comparisons/index.md

```diff
@@ -0,0 +1,562 @@
+
+#### Modules and Libraries
+
+##### Node.js
+
+Node.js computer language is written in a unique structure. One characteristic to consider is the steps to create or add modules into Node.js. As each coding language includes its sequenced procedure, Node.js also involves flexible options.  
+
+We can find Node.js modules on open-source websites. In Node.js, folders and JavaScript files are known to hold modules. Modules inside the `node_module` folder can be used in the Node.js application.  
+
+A branched-out path directory looks similar to the picture below.  
+
+[directory](/engineering-education/nodejs-structural-comparisons/path.jpg)  
+_Screen capture_  
+
+Remember to use the `npm rebuild` command.  
+
+```JavaScript
+global.puny = require('punycode')
+```  
+
+Functions for the module should appear at the bottom of the Node.js application window.  
+
+The choice to include dependencies is usually not available. This module can be used with the codes shown below.  
+
+```JavaScript
+global.char = 'sectionio';
+global.changeto = puny.decode(char);
+```  
+
+**Output:**  
+```JavaScript
+'ઽ઺ૃરીૄ૆'
+```  
+
+The code snippet below shows how to encode the above output back to the original text string variable.  
+
+```JavaScript
+puny.encode(changeto);
+```  
+
+**Output:**  
+```JavaScript
+'sectionio'
+```  
+
+Shown below is how the code is displayed in a Node.js window.  
+
+[punycode code snippet](/engineering-education/nodejs-structural-comparisons/punycodenodejs.jpg)  
+_Screen capture_  
+
+As a side note, installing external Node.js modules can be found in the documentation.  
+
+##### R-Programming
+
+More information can be found in [CRAN documentation](https://cran.r-project.org). Further options can include `dependencies =` and `source = `.  
+
+Similar to Node.js, R-Programming also uses browser windows to display. Usually, in R-Programming, RStudio can display visuals. However, Graphic User Interface (GUI) applications are sent to an internet browser window.  
+
+##### C/C++
+
+The following is the most frequently used procedure to install a C module. An advanced statistical module will be installed. More information on how to install the TensorFlow library can be found [here](https://www.tensorflow.org/install/lang_c).  
+
+1. Find a module.
+2. Copy the URL.
+3. Enter in the following command:  
+```Bash
+wget https://storage.googleapis.com/tensorflow/libtensorflow/libtensorflow-cpu-linux-x86_64-2.4.0.tar.gz
+```  
+
+4. Extract. Usually with:  
+```Bash
+tar -C /usr/local -xzf libtensorflow-cpu-linux-x86_64-2.4.0.tar.gz
+```  
+6. Enter:  
+```Bash
+ldconfig
+```  
+Visuals of this procedure can be found [here](https://www.analyticsvidhya.com/blog/2021/04/alternative-tools-for-effective-machine-learning/).  
+
+##### Java
+
+This programming language can be limited. As NetBeans can attempt to bring a vast number of features, the portability and compatibility are limited as technological preferences change.  
```

Comment:
As far as I know, Java is one of the most robust and feature rich languages. Why do you call it limited? Add more content to the explanation.

Reply:
I understand Java is robust, however, most users would not choose Java to perform certain concepts like graphing tasks as some functions cannot support their preferences. I can change the sentence to specify the why and what. More about preferences for speed and graphing components.